File: app/src/main/java/com/samourai/wallet/util/view/ViewHelper.java
package com.samourai.wallet.util.view;

import android.animation.ArgbEvaluator;
import android.animation.ValueAnimator;
import android.view.View;

public class ViewHelper {

    private ViewHelper() {
    }

    public static void animateChangeBackgroundColor(
            final View view,
            final int colorFrom,
            final int colorTo,
            final int durationInMs) {

        final ValueAnimator colorAnimation = ValueAnimator.ofObject(
                new ArgbEvaluator(),
                colorFrom,
                colorTo);
        colorAnimation.setDuration(durationInMs);
        colorAnimation.addUpdateListener(animator -> view.setBackgroundColor((int) animator.getAnimatedValue()));
        colorAnimation.start();
    }

    public static void animateChangeBackgroundColor(
            final ValueAnimator.AnimatorUpdateListener listener,
            final int colorFrom,
            final int colorTo,
            final int durationInMs) {

        final ValueAnimator colorAnimation = ValueAnimator.ofObject(
                new ArgbEvaluator(),
                colorFrom,
                colorTo);
        colorAnimation.setDuration(durationInMs);
        colorAnimation.addUpdateListener(listener);
        colorAnimation.start();
    }
}

File: app/src/main/java/com/samourai/wallet/util/view/ViewHelper.kt
package com.samourai.wallet.util.view

import android.content.Context
import androidx.compose.ui.unit.Density
import androidx.compose.ui.unit.Dp

class ViewHelper {

    companion object {
        fun convertDpToPx(valueInDp: Dp, density: Density): Float {
            return with(density) { valueInDp.toPx() }
        }

        fun getScreenWidthPx(context: Context): Int {
            val resources = context.resources
            val displayMetrics = resources.displayMetrics
            return displayMetrics.widthPixels
        }
    }
}